### PR TITLE
Feature/issue 310

### PR DESCRIPTION
**Pull request by hideki**
1. Generate uppercase Hex string from byte array
2. Check lower case file name for previous version of CBL Android/Java

### CODE REVIEW

**Comment by hideki** [comment]
HI @pasin,

I added following changes that we discussed on Monday with @zgramana. Could you please review this PR too?

- If CBL found the lowercase attachment filename, show error or warning message.
- Manager should have option for auto migrate for attachment filename.
- If auto migration for attachment filename is TRUE, change all filenames from lowercase filename to uppercase filename.
- If migration is done, store the flag into local doc.

File: src/main/java/com/couchbase/lite/Database.java [hunk 1]
```diff
@@ -1125,6 +1130,22 @@ public synchronized boolean open() {
         return true;
     }
 
+    private boolean isBlobstoreMigrated(){
+        Map<String, Object> props =getExistingLocalDocument("_blobstore");
```
Comment (pasin) [inline src/main/java/com/couchbase/lite/Database.java]: Code Style : Space after =

File: src/main/java/com/couchbase/lite/Database.java [hunk 2]
```diff
@@ -1125,6 +1130,22 @@ public synchronized boolean open() {
...
+        if(props != null && props.containsKey("blobstoreMigrated"))
+            return (Boolean)props.get("blobstoreMigrated");
+        return false;
+    }
+    private void markBlobstoreMigrated(){
```
Comment (pasin) [inline src/main/java/com/couchbase/lite/Database.java]: Code Style: Space after()

File: src/main/java/com/couchbase/lite/Manager.java [hunk 3]
```diff
@@ -664,6 +665,12 @@ public Context getContext() {
         return context;
     }
 
+    public boolean isAutoMigrateBlobStoreFilename() {
+        return autoMigrateBlobStoreFilename;
+    }
 
+    public void setAutoMigrateBlobStoreFilename(boolean autoMigrateBlobStoreFilename) {
```
Comment (pasin) [inline src/main/java/com/couchbase/lite/Manager.java]: Should we put this into ManagerOption? What is your opinion?

**Comment by hideki** [comment]
@pasin I applied feedbacks from you. Please review again.

**Comment by hideki** [comment]
@pasin I applied your feedback. Please review again.

**Pull request merged**